<?xml version="1.0"?>
<launch>
  <arg name="chain_start" default="base_link" />
  <arg name="chain_end" default="moveo_nozzle_link" />
  <arg name="timeout" default="0.03" />
  <arg name="eps" default="1e-5"/>
  <arg name="num_threads" default="3"/>
  <arg name="model" default="$(find moveo_urdf)/urdf/moveo_urdf.urdf"/>
  <param name="robot_description" textfile="$(arg model)" />
  
  <arg name="gcode_in" default="$(find gcode_translation)/gcode/new split 15mm test.gcode"/>
  <arg name="gcode_out" default="$(find gcode_translation)/gcode/new 15mm test1.gcode"/>

  <node name="inverse_kinematics_origin" pkg="gcode_translation" type="inverse_kinematics_origin" output="screen">
    <param name="chain_start" value="$(arg chain_start)"/>
    <param name="chain_end" value="$(arg chain_end)"/>
    <param name="timeout" value="$(arg timeout)"/>
    <param name="eps" value="$(arg eps)" type = "double"/>
    <param name="num_threads" value="$(arg num_threads)" type = "int"/>
    <param name="urdf_param" value="/robot_description"/>
    <param name="gcode_in" value="$(arg gcode_in)"/>
    <param name="gcode_out" value="$(arg gcode_out)"/>
  </node>
</launch>
